### DISCUSSION TITLE
Lazy Load clarity tree

### DISCUSSION THREAD

**Discussion by jeevanm18** [vmware-clarity/ng-clarity#267] [General]
Unable to Lazy load clr tree children inside clr tree on clicking a button

<clr-tree>
                <clr-tree-node [clrExpanded]="true">
                    <div ngProjectAs="clr-tree-node" *clrIfExpanded>
                        <clr-tree-node *ngFor="let node of getChildren">
                            <button (click)="nodeSelected(node)" class="clr-treenode-link" [class.active]="node.active" [class.text-muted]="node.disabled" [disabled]="node.disabled">
                                {{node.name}}
                            </button>
                        </clr-tree-node>

                        <clr-tree-node>
                            <button class="btn btn-sm" *ngIf="limitAt === 3" (click)="limitAt = 6">Load More...</button>
                            <button class="btn btn-sm" *ngIf="limitAt === 6" (click)="limitAt = 9">Load Even More...</button>
                        </clr-tree-node>
                    </div>
                </clr-tree-node>
            </clr-tree>

 get getChildren() {
        if (this.nodedata.children.children.length > 7) {
            let data = this.nodedata.children[6].children.slice(0, 5);
            return data;
        }
        return this.nodedata.children;

**Comment by kevinbuhmann** [comment]
See #207. I posted a proof-of-concept that may be helpful. Essentially, you need to break reference for the collection passed to `*clrRecursiveFor`.
